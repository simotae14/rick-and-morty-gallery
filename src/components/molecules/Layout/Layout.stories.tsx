import type { Meta, StoryObj } from '@storybook/react';

import Layout from './index';

const meta = {
  title: 'Molecules/Layout',
  component: Layout,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof Layout>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: <div>content</div>,
  },
};
