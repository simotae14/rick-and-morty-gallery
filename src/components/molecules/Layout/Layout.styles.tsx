import styled from 'styled-components';

export const BaseWrapper = styled.div`
  display: flex;
  max-width: 1220px;
  margin: 0 auto;
`;
