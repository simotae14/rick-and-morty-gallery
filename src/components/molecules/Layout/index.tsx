import { BaseWrapper } from './Layout.styles';

/**
 * Component used for the Shared layout
 */
function Layout({
  children,
}: {
  /**
   * It is the content wrapped inside the layout
   */
  children: React.ReactNode;
}) {
  return <BaseWrapper>{children}</BaseWrapper>;
}

export default Layout;
