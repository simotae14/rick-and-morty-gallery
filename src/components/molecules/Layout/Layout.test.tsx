/* eslint-disable import/no-extraneous-dependencies */
import { describe, it } from 'vitest';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';

import Layout from '.';

expect.extend(toHaveNoViolations);

describe('Components => molecules => Layout', () => {
  it('should render the Layout', () => {
    const { asFragment } = render(<Layout>content</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the content inside the Layout', () => {
    const { getByText } = render(<Layout>content</Layout>);
    expect(getByText(/content/i)).toBeInTheDocument();
  });

  it('should not fail any accessibility tests', async () => {
    const { container } = render(<Layout>content</Layout>);
    expect(await axe(container)).toHaveNoViolations();
  });
});
