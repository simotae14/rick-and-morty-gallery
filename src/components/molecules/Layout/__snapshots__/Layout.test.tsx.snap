// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components => molecules => Layout > should render the Layout 1`] = `
<DocumentFragment>
  <div
    class="sc-eLqIir dvIhby"
  >
    content
  </div>
</DocumentFragment>
`;
